bin/zis-parmlib-update.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/sh
+# This program and the accompanying materials are
+# made available under the terms of the Eclipse Public License v2.0 which accompanies
+# this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright Contributors to the Zowe Project.
+
+# This script adds or updates pramaters in ZIS PARMLIB member specified in
+# the provided SAMPLIB file.
+
+# Make sure all files created in this script are only accesible by the owner.
+umask go-rwx
+
+# Prohibit undefined variables.
+set -u
+
+# ZIS-related global variables.
+ZIS_PREFIX=${ZIS_PREFIX:-$USER.DEV}
+ZIS_PARMLIB=${ZIS_PARMLIB:-$ZIS_PREFIX.PARMLIB}
+ZIS_PARMLIB_MEMBER=${ZIS_PARMLIB_MEMBER:-ZWESIP00}
+
+#################################################
+# Generate a name for a temporary file in /tmp
+#
+# Globals:
+#   RANDOM
+# Arguments:
+#   None
+# Returns:
+#   The generated file name
+#################################################
+make_temp_filename() {
+  name=/tmp/zis-sample-install-$$-$RANDOM
+  while [ -e $name ]
+  do
+    name=/tmp/zis-sample-install-$$-$RANDOM
+  done
+  echo $name
+}
+
+#################################################
+# Try to resolve values that are defined using
+# environmental variables, otherwise return
+# the original value
+#
+# Globals:
+#   None
+# Arguments:
+#   $1 Value from a key-value pair
+# Returns:
+#   * If an env variable is provided, its value
+#     is returned on success
+#   * If an env varible if provided and
+#     the variable is not defined,
+#     string VALUE_NOT_FOUND is returned
+#   * The original value is returned
+#################################################
+resolve_env_parameter() {
+
+  parm=$1
+
+  # Are we dealing with an env variable based value?
+  echo $parm | grep -E "^[$]{1}[A-Z0-9_]+$" 1>/dev/null
+  # Yes, resolve the value using eval, otherwise return the value itself.
+  if [ $? -eq 0 ]; then
+    ( eval "echo $parm" 2>/dev/null )
+    if [ $? -ne 0 ]; then
```

Comment:
what is this supposed to do?

Reply:
Yes, it will.

Reply:
I tried: it did not. Why would it? `echo $parm` expands to just `echo` which just echos an empty string and exits.

Reply:
Did you issue ```set -u```? It'll cause an error if an undefined variable is being dereferenced.

Reply:
Ah, ok. Though, I'm still not sure why we need two nested subshells: the parenthesis are one, and the `eval` is another. `(set -u && echo $foo) 2>/dev/null` seems to be enough

Reply:
Let's say ```$parm='$FOO'```. ```echo``` wouldn't be able to resolve ```'$FOO'```, it would just print the name of the variable, i.e. you'd see ```$FOO```, not the value behind FOO.